core/src/main/java/nl/nn/adapterframework/receivers/ReceiverBase.java

```diff
@@ -1033,28 +1035,34 @@ public void retryMessage(String messageId) throws ListenerException {
 		TransactionStatus txStatus = itx.getStatus();
 		Serializable msg=null;
 		ITransactionalStorage<Serializable> errorStorage = getErrorStorage();
+		threadContext.put("retry", "true");
 		try {
 			try {
-				msg = errorStorage.getMessage(messageId);
+				msg = errorStorage.getMessage(storageKey);
 				processRawMessage(msg, threadContext, -1, true);
 			} catch (Throwable t) {
 				txStatus.setRollbackOnly();
 				throw new ListenerException(t);
 			} finally {
-				txManager.commit(txStatus);
+				if (txStatus.isRollbackOnly()) {
+					txManager.rollback(txStatus);
+				} else {
+					txManager.commit(txStatus);
```

Comment:
Eigenlijk zou ik verwachten dat de txManager slimmer zou zijn, dwz dat hij in zijn commit methode eerst controlleerd of hij marked as rollback is.

Reply:
Dat doet hij ook wel, maar als dat dan zo is, dan rolt hij terug en zegt dat de rollback unexpected was